experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -410,56 +408,28 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
 	}
 
 	/**
-	 * Updates paths for a range move (including re-parenting path items and updating indexes).
-	 * @param count - number of siblings to insert/delete/move.
-	 * @param srcStart - where the siblings are removed from. If undefined the operation is an insert.
-	 * @param dst - where the siblings are moved to. If undefined the operation is a delete.
-	 *
-	 * TODO:
-	 * How should anchors that become invalid, then valid again (ex: into content that was deleted, then undone) work?
-	 * Add an API to resurrect them? Store them in special detached fields? Store them in special non-detached fields?
-	 *
-	 * TODO:
-	 * How should custom anchors work (ex: ones not just tied to a specific Node)?
-	 * This design assumes they can be expressed in terms of a Node anchor + some extra stuff,
-	 * but we don't have an API for the extra stuff yet.
+	 * Detach nodes from their parent.
+	 * @param count - number of siblings that are detached from the original tree.
+	 * @param src - where the siblings are detached from.
 	 *
 	 * TODO: tests
 	 */
-	private moveChildren(
-		count: number,
-		srcStart: UpPath | undefined,
-		dst: UpPath | undefined,
-	): void {
-		assert(
-			srcStart !== undefined || dst !== undefined,
-			0x352 /* moveChildren is a no-op and should not be called if there is no src or dst */,
-		);
-
-		const srcParent =
-			srcStart === undefined ? undefined : this.find(srcStart.parent ?? this.root);
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		const srcChildren = srcParent?.children?.get(srcStart!.parentField);
-		// Sorted list of PathNodes to move from src to dst.
-		let toMove: PathNode[];
+	private detachNodes(count: number, src: UpPath): PathNode[] {
```

Comment:
usually we do path then count: I think that order makes more sense since the count is relative to the path and we generally put more significant/ large scoped parameters first.